src/Common/TagStoreInterface.php

```diff
@@ -4,15 +4,73 @@
 
 interface TagStoreInterface
 {
-    public function tagObject($objectId, array $tags);
+    /**
+     * Associate the given tags to cache item identified by $cacheId.
+     *
+     * Example: associate a HTTP cache identifier (e.g. a content-digest in the
+     *          case of the Symfony HTTP cache) with a set of tags.
+     *
+     * @param integer $cacheId
+     * @param string[] $tags
+     */
+    public function tagCacheId($cacheId, array $tags);
 
-    public function untagObject($objectId, array $tags);
+    /**
+     * Unassociate the given tags from the cache item identified by $cacheId.
+     *
+     * NOTE: This is difficult for an so-called "cache based" implementation, is it necessary?
+     *
+     * @param integer $cacheId
+     * @param string[] $tags
+     */
+    public function untagCacheId($cacheId, array $tags);
```

Comment:
In a "cache-based" implementation we would store the `tag` as the `key`, and the `cache identifiers` would be serialized in the cache entry, therefore searching based on cache-ids is not efficient here.

What would the use case be for this method?